### PR TITLE
#345 | Admin session handling improved

### PR DESCRIPTION
**Pull request by szymon-owczarzak**
## Description

- Admin Session extended to 5 hours
- Admin Session will be refreshed on every POST and DELETE request.

## Motivation and Context

It was quite easy to lost changes made to boards due to session expiration.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](https://github.com/wttech/cogboard/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [x] Automated functional tests have been added or modified to cover my changes (if applicable)
- [ ] I have updated the documentation accordingly.

---

I hereby agree to the terms of the Cogboard Contributor License Agreement.

### CODE REVIEW

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/handler/HandlerUtil.kt [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
-                    .end(body)
+            if (SESSION_REFRESHERS.contains(routingContext.request().method())) {
+                routingContext.request().headers().add("body", body)
+                routingContext.reroute(HttpMethod.POST, "/api/session/refresh")
+            } else routingContext.response()
+                .putHeader("Content-Type", "application/json")
+                .end(body)
         }
+
+        val SESSION_REFRESHERS = setOf(HttpMethod.POST, HttpMethod.DELETE)
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/handler/HandlerUtil.kt:17]: All `POST` and `DELETE` should now refresh the session

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/JwtCommon.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.cognifide.cogboard.utils.ExtensionFunctions.endEmptyJson
+import com.cognifide.cogboard.utils.ExtensionFunctions.toJWT
+import io.vertx.core.json.JsonObject
+import io.vertx.ext.auth.KeyStoreOptions
+import io.vertx.ext.jwt.JWTOptions
+import io.vertx.reactivex.core.Vertx
+import io.vertx.reactivex.ext.auth.jwt.JWTAuth
+import io.vertx.reactivex.ext.web.RoutingContext
+
+open class JwtCommon {
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/JwtCommon.kt:13]: Common logic for handling JWT token

File: cogboard-webapp/src/actions/helpers.js [hunk 3]
```diff
@@ -29,7 +29,12 @@ export const fetchData = (
 
   return fetch(url, authenticationConfig)
     .then(checkResponseStatus)
-    .then(response => response.json());
+    .then(response => {
+      if (response.headers.get('token')) {
+        setToken(response.headers.get('token'));
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/actions/helpers.js:34]: Any API response can potentially contain refreshed token

File: cogboard-webapp/src/actions/thunks.js [hunk 4]
```diff
@@ -91,8 +91,7 @@ export const login = (credentials, loginAsGuest) => dispatch => {
     dispatch(pushNotification(NOTIFICATIONS.LOGIN(`Guest: ${guestName}`)));
   } else {
     return fetchData(URL.LOGIN, { method: 'POST', data: credentials }).then(
-      ({ token }) => {
-        setToken(token);
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/actions/thunks.js:95]: moved to `fetchData`

File: functional/cypress-tests/cypress/support/dashboards.js [hunk 5]
```diff
@@ -66,7 +66,7 @@ Cypress.Commands.add(
       username: username,
       password: password
     }).then(response => {
-      const loginBody = response.body.token.split(' ');
+      const loginBody = response.headers['token'].split(' ');
```
Comment (szymon-owczarzak) [inline functional/cypress-tests/cypress/support/dashboards.js:69]: the token is now returned as a header - as it should be from the begining

**Review by devzbysiu — APPROVED** [review]

**Review by marcin-witaszek — APPROVED** [review]

**Pull request merged**